reports/pin/pin.qmd

```diff
@@ -15,23 +15,364 @@ format:
     fontsize: 12pt
 editor: source
 params:
-  run_id: "2024-01-10-objective-patrick"
+  run_id: "2024-01-07-great-ida"
   year: "2024"
-  pin: "13253180150000"
+  pin:
+    - "16083110070000"
 ---
 
 {{< include ../_setup.qmd >}}
 
-```{r pin_checks}
-target_pin <- ccao::pin_clean(params$pin)
+```{r _pin_checks}
+target_pins <- ccao::pin_clean(params$pin)
 
-if (nchar(target_pin) != 14) {
-  stop("Invalid PIN: must be 14 characters long")
+for (target_pin in target_pins) {
+  if (nchar(target_pin) != 14) {
+    stop(paste("Invalid PIN:", target_pin, "must be 14 characters long"))
+  }
+
+  if (!target_pin %in% assessment_card$meta_pin) {
+    stop(paste("Invalid PIN:", target_pin, "not found in assessment data"))
+  }
 }
+```
 
-if (!target_pin %in% assessment_card$meta_pin) {
-  stop("Invalid PIN: not found in assessment data")
+```{r _pin_URLs, results='asis'}
+base_url <- "https://www.cookcountyassessor.com/pin/"
+pin_values <- params$pin
+
+for (pin in pin_values) {
+  full_url <- paste0(base_url, pin)
+  cat(sprintf('<a href="%s">%s</a><br/>', full_url, full_url))
 }
 ```
 
-This is a stub report for the PIN `r target_pin` and run ID `r params$run_id`!
+```{r _pin_setup}
+noctua_options(cache_size = 10)
+con <- dbConnect(noctua::athena())
+
+pins_clause <- paste0("('", paste(params$pin, collapse = "', '"), "')")
+
+res_chars <- dbGetQuery(
+  conn = con,
+  sprintf("SELECT pin, year, pin10, pin_is_multicard, card, class, char_yrblt, char_bldg_sf, char_beds, char_fbath, char_hbath, char_air, char_type_resd, char_rooms FROM default.vw_card_res_char WHERE CAST(year AS INTEGER) >= %d AND pin IN %s", as.numeric(params$year) - 3, pins_clause)
+)
+
+pin_value <- dbGetQuery(
+  conn = con,
+  sprintf("SELECT * FROM default.vw_pin_value WHERE CAST(year AS INTEGER) >= %d AND pin IN %s", as.numeric(params$year) - 3, pins_clause)
+)
+
+pin_sale <- dbGetQuery(
+  conn = con,
+  sprintf("SELECT year, pin, num_parcels_sale, sv_outlier_type, doc_no, sale_date, sale_price FROM default.vw_pin_sale WHERE CAST(year AS INTEGER) >= %d AND pin IN %s", as.numeric(params$year) - 10, pins_clause)
+)
+
+hie_data_subset <- hie_data %>%
+  filter(hie_last_year_active >= params$year) %>%
+  mutate(hie = TRUE)
+```
+
+## Predicted Model Value
+
+```{r, _pin_predicted_model_value}
+assessment_pin %>%
+  mutate(township = ccao::town_convert(meta_township_code)) %>%
+  filter(meta_pin %in% params$pin) %>%
+  arrange(meta_pin) %>%
+  select(
+    "Township" = township,
+    "Final FMV" = pred_pin_final_fmv
+  ) %>%
+  mutate(`Final FMV` = scales::dollar(`Final FMV`)) %>%
+  kable() %>%
+  kable_styling(
+    "striped",
+    position = "left",
+    fixed_thead = TRUE,
+    html_font = "Consolas"
+  )
+```
```

Comment:
**issue (blocking):** The pipeline actually produces multiple levels of predictions to show each step in the post-processing pipeline. I would include those values in this table, look for every field in `assessment_pin` that starts with `pred_`. Also, the final value we send to folks is `pred_pin_final_fmv`.